packages/ember-metal/lib/is_empty.ts

```diff
@@ -52,18 +52,16 @@ export default function isEmpty(obj: any): boolean {
     if (typeof size === 'number') {
       return !size;
     }
-  }
-
-  if (typeof obj.length === 'number' && objectType !== 'function') {
-    return !obj.length;
-  }
 
-  if (objectType === 'object') {
     let length = get(obj, 'length');
     if (typeof length === 'number') {
       return !length;
     }
   }
 
+  if (typeof obj.length === 'number' && objectType !== 'function') {
```

Comment:
Hmm, do we even need this check anymore?

Reply:
This is what allows an empty string (`''`) to be marked as empty.